chainer/training/triggers/interval.py

```diff
@@ -40,9 +42,16 @@ def __call__(self, trainer):
         """
         updater = trainer.updater
         if self.unit == 'epoch':
-            prev = self.count
-            self.count = updater.epoch_detail // self.period
-            return prev != self.count
+            epoch_detail = updater.epoch_detail
+            previous_epoch_detail = updater.previous_epoch_detail
```

Comment:
> >     Nodes in the abstract computational graph correspond to chainer.Variables and operators to chainer.Functions.
>
> 
> I understand some nodes are not called for each iteration. However, from my understanding, Variable and Function do not have any Trainer objects. How do they call a trigger internally?

This is outside the standard features of `Variable(Node)` and `Function`, but an extension does this.

In an extension I'm developing, it holds "logical" variable and function nodes(*) and triggers associated with them. Not necessarily all logical nodes are instantiated in each iteration, because of Define-by-Run nature. Only when the associated logical nodes are instantiated, the extension call these triggers with the trainer instance (which are given to the extension).

(*) Corresponding variables and functions across multiple iterations are considered a single "logical" variables and functions, despite distinct physical `Variable(Node)` and `Function` objects.